src/frontend/planner_test/tests/testdata/nexmark_source.yaml

```diff
@@ -400,27 +400,25 @@
       └─StreamFilter { predicate: (count >= max(count)) }
         └─StreamHashJoin { type: Inner, predicate: window_start = window_start, output: all }
           ├─StreamExchange { dist: HashShard(window_start) }
-          | └─StreamProject { exprs: [auction, window_start, count] }
-          |   └─StreamShare { id = 7 }
-          |     └─StreamAppendOnlyHashAgg { group_key: [auction, window_start], aggs: [count] }
-          |       └─StreamExchange { dist: HashShard(auction, window_start) }
-          |         └─StreamHopWindow { time_col: date_time, slide: 00:00:02, size: 00:00:10, output: [auction, window_start, _row_id] }
-          |           └─StreamProject { exprs: [auction, date_time, _row_id] }
-          |             └─StreamFilter { predicate: IsNotNull(date_time) }
-          |               └─StreamRowIdGen { row_id_index: 7 }
-          |                 └─StreamSource { source: "bid", columns: ["auction", "bidder", "price", "channel", "url", "date_time", "extra", "_row_id"] }
+          | └─StreamShare { id = 7 }
```

Comment:
An observation: this plan can be further optimized because all parents of the `share` node are `StreamExchange { dist: HashShard(window_start) }`. Currently, we need to shuffle twice, but one of them is wasted.

Reply:
Sounds we need a better `StreamShare::to_stream()`

Reply:
Yes, to remove this shuffle, we need to collect some information before `StreamShare::to_stream()`.

Reply:
First I guess we can make all `Share` transform to `NoShuffle` exchange, but there seem to be some bugs in scaling now. I'll investigate it.